### PR TITLE
Whack a mole

### PR DESCRIPTION
**Pull request by AlekseyGrimm**
https://github.com/rolling-scopes-school/tasks/blob/master/tasks/stage-0/projects.md#task-10-whack-a-mole-40

### CODE REVIEW

File: Whack-a-mole/index.html [hunk 1]
```diff
@@ -0,0 +1,37 @@
+<html lang="en">
+
+<head>
+    <meta charset="UTF-8">
+    <title>Whack A Mole!</title>
+    <link rel="stylesheet" href="style.css">
+</head>
+
+<body>
+    <h1>Whack-a-mole!<span class="score">11</span></h1>
```
Comment (y-solovey) [inline Whack-a-mole/index.html]: please set the initial value to 0

File: Whack-a-mole/style.css [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+.hole {
+  flex: 1 0 33.33%;
+  overflow: hidden;
+  position: relative;
+}
+
+.hole:after {
+  display: block;
+  background: url(https://js3030.github.io/dirt.svg) bottom center no-repeat;
```
Comment (y-solovey) [inline Whack-a-mole/style.css]: please move pictures to your repository

File: Whack-a-mole/index.html [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        </div>
+        <div class="hole hole5">
+            <div class="mole"></div>
+        </div>
+        <div class="hole hole6">
+            <div class="mole"></div>
+        </div>
+    </div>
+    <script src="main.js"></script>
+    <div class="mallbery-caa" style="z-index: 2147483647 !important; text-transform: none !important; position: fixed;">
```
Comment (y-solovey) [inline Whack-a-mole/index.html]: please remove this container

File: Whack-a-mole/main.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+function randomTime(min, max) {
+    return Math.round(Math.random() * (max - min) + min);
+}
+
+function randomHole(holes) {
+    var idx = Math.floor(Math.random() * holes.length);
+    var hole = holes[idx];
+    if (hole === lastHole) {
+        console.log("Ah nah thats the same one bud");
```
Comment (y-solovey) [inline Whack-a-mole/main.js]: please remove logging

File: Whack-a-mole/style.css [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+*,
+*before,
+*after {
+  box-sizing: inherit;
+}
+
+body {
+  padding: 0;
+  margin: 0;
+  font-family: "Amatic SC", cursive;
```
Comment (y-solovey) [inline Whack-a-mole/style.css:22]: please import the font

File: Whack-a-mole/main.js [hunk 6]
```diff
@@ -0,0 +1,48 @@
+var holes = document.querySelectorAll(".hole");
```
Comment (y-solovey) [inline Whack-a-mole/main.js]: please change to const and let

File: Whack-a-mole/main.js [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+function startGame() {
+    scoreBoard.textContent = 0;
+    timeUp = false;
+    score = 0;
+    peep();
+    setTimeout(() => timeUp = true, 10000)
+}
+
+function bonk(e) {
+    if (!e.isTrusted) return; // cheater!
```
Comment (y-solovey) [inline Whack-a-mole/main.js]: what does this code do?

**Review by y-solovey — CHANGES_REQUESTED** [review]

File: Whack-a-mole/index.html (hunk 1 above)
Comment (AlekseyGrimm) [reply Whack-a-mole/index.html]: done

File: Whack-a-mole/style.css (hunk 2 above)
Comment (AlekseyGrimm) [reply Whack-a-mole/style.css]: done

File: Whack-a-mole/index.html (hunk 3 above)
Comment (AlekseyGrimm) [reply Whack-a-mole/index.html]: well done

File: Whack-a-mole/main.js (hunk 4 above)
Comment (AlekseyGrimm) [reply Whack-a-mole/main.js]: ops...
removed this log

File: Whack-a-mole/style.css (hunk 5 above)
Comment (AlekseyGrimm) [reply Whack-a-mole/style.css:22]: done

File: Whack-a-mole/main.js (hunk 6 above)
Comment (AlekseyGrimm) [reply Whack-a-mole/main.js]: done

File: Whack-a-mole/main.js (hunk 7 above)
Comment (AlekseyGrimm) [reply Whack-a-mole/main.js]: ops)) I deleted this code

File: Whack-a-mole/style.css (hunk 5 above)
Comment (y-solovey) [reply Whack-a-mole/style.css:22]: please import the font with `@font-face`

**Review by y-solovey — CHANGES_REQUESTED** [review]

**Review by y-solovey — APPROVED** [review]

File: Whack-a-mole/main.js [hunk 8]
```diff
@@ -0,0 +1,47 @@
+const holes = document.querySelectorAll(".hole");
+const scoreBoard = document.querySelector(".score");
+const moles = document.querySelectorAll(".mole");
+let lastHole;
+let timeUp = false;
+let score = 0;
+
```
Comment (y-solovey) [inline Whack-a-mole/main.js:17]: According to https://github.com/rolling-scopes-school/tasks/blob/master/tasks/stage-0/projects.md#task-10-whack-a-mole-40, please add local storage saving and progressively more difficult levels

**Review by y-solovey — CHANGES_REQUESTED** [review]

File: Whack-a-mole/style.css (hunk 5 above)
Comment (AlekseyGrimm) [reply Whack-a-mole/style.css:22]: done

File: Whack-a-mole/main.js [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+const scoreBoard = document.querySelector(".score");
+const moles = document.querySelectorAll(".mole");
+const level1 = document.querySelector(".easy");
+const level2 = document.querySelector(".medium");
+const level3 = document.querySelector(".hard");
+const record = document.querySelector(".record");
+let lastHole;
+let timeUp = false;
+let score = 0;
+let rec = localStorage.getItem("totalScore");;
```
Comment (y-solovey) [inline Whack-a-mole/main.js]: initial value can be undefined so set it to 0 in this case

File: Whack-a-mole/main.js [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+// writes the score
+function bonk(e) {
+    if (!e.isTrusted) return;
+    score++;
+    this.classList.remove("up");
+    scoreBoard.textContent = score;
+    record.textContent = rec;
+    if (score >= rec) { 
+        rec = score;
+        rec ++
```
Comment (y-solovey) [inline Whack-a-mole/main.js]: why we add `rec` here?

File: Whack-a-mole/index.html [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+
+<head>
+    <meta charset="UTF-8">
+    <title>Whack A Mole!</title>
+    <link rel="stylesheet" href="style.css">
+</head>
+
+<body>
+    <h1>Whack-a-mole!<span class="score">0</span></h1>
+    <h2>Record!<span class="record">0</span></h2>
```
Comment (y-solovey) [inline Whack-a-mole/index.html]: we need to update `records` if we have the value in a local storage

Comment (y-solovey) [reply Whack-a-mole/index.html]: Please improve label as "Record: X" where X is 0 or the value from the local storage

**Review by y-solovey — CHANGES_REQUESTED** [review]

File: Whack-a-mole/main.js [hunk 12]
```diff
@@ -0,0 +1,81 @@
...
+const scoreBoard = document.querySelector(".score");
+const moles = document.querySelectorAll(".mole");
+const level1 = document.querySelector(".easy");
+const level2 = document.querySelector(".medium");
+const level3 = document.querySelector(".hard");
+const record = document.querySelector(".record");
+let lastHole;
+let timeUp = false;
+let score = 0;
+let recording = localStorage.getItem("totalScore");
```
Comment (y-solovey) [inline Whack-a-mole/main.js]: you need a value for initial loading when locaStorage is empty

File: Whack-a-mole/main.js [hunk 13]
```diff
@@ -0,0 +1,81 @@
...
+
+
+window.onload = function (){
+    record.textContent = recording;
+    recording.textContent = localStorage.getItem("totalScore");
```
Comment (y-solovey) [inline Whack-a-mole/main.js]: what does this code do?

File: .vscode/settings.json [hunk 14]
```diff
@@ -0,0 +1,3 @@
+{
+    "liveServer.settings.port": 5501
```
Comment (y-solovey) [inline .vscode/settings.json]: please remove this file

**Review by y-solovey — CHANGES_REQUESTED** [review]

File: Whack-a-mole/main.js (hunk 8 above)
Comment (AlekseyGrimm) [reply Whack-a-mole/main.js:17]: done

File: Whack-a-mole/main.js (hunk 9 above)
Comment (AlekseyGrimm) [reply Whack-a-mole/main.js]: done

File: Whack-a-mole/main.js (hunk 10 above)
Comment (AlekseyGrimm) [reply Whack-a-mole/main.js]: update code

File: .vscode/settings.json (hunk 14 above)
Comment (AlekseyGrimm) [reply .vscode/settings.json]: done

File: Whack-a-mole/main.js (hunk 13 above)
Comment (AlekseyGrimm) [reply Whack-a-mole/main.js]: This code loading value localstorage when  load page. 
I'm update this code 
window.onload = function (){
    record.textContent = localStorage.getItem("totalScore");
 };

File: Whack-a-mole/main.js (hunk 12 above)
Comment (AlekseyGrimm) [reply Whack-a-mole/main.js]: done

Comment (y-solovey) [reply Whack-a-mole/main.js]: typically use such expression `let recording = localStorage.getItem("totalScore") || 0;`

File: Whack-a-mole/main.js (hunk 13 above)
Comment (y-solovey) [reply Whack-a-mole/main.js]: Does `recording` have `textContent`?

**Review by y-solovey — CHANGES_REQUESTED** [review]

File: Whack-a-mole/index.html (hunk 11 above)
Comment (AlekseyGrimm) [reply Whack-a-mole/index.html]: done

**Review by y-solovey — APPROVED** [review]

**Pull request merged**